### PR TITLE
Add an HTTP wrapper to the gRPC daemon

### PR DESCRIPTION
**Pull request by tomelm**
### Reviewers
r? @vcheung-stripe @suz-stripe 
cc @stripe/developer-products

 ### Summary
I was working on a small side projects where I was trying to have an app talk to the Stripe CLI's gRPC interface. One problem I had is that the request client only supported HTTP which meant that it couldn't interface with the daemon directly. This PR wraps the gRPC daemon and spins up another server that supports all the gRPC functionality as an HTTP server.

### CODE REVIEW

**Review by ianjabour-stripe — APPROVED** [review]
This LGTM, will defer to @vcheung-stripe being that he's got more context on the gRPC setup.

File: pkg/rpcservice/middleware.go [hunk 1]
```diff
@@ -43,7 +44,10 @@ func authorize(ctx context.Context) error {
 	}
 
 	if _, ok := md[requiredHeader]; !ok {
-		return status.Errorf(codes.Unauthenticated, fmt.Sprintf("%s header is not supplied", requiredHeader))
+		// Check for the alt header too, which is required for grpc web clients
+		if _, ok := md[altRequiredHeader]; !ok {
+			return status.Errorf(codes.Unauthenticated, fmt.Sprintf("%s header is not supplied", requiredHeader))
```
Comment (vcheung-stripe) [inline pkg/rpcservice/middleware.go:49]: nit: do you want to return the alt header in this case?
```suggestion
			return status.Errorf(codes.Unauthenticated, fmt.Sprintf("%s header is not supplied", altRequiredHeader))
```

**Review by vcheung-stripe — APPROVED** [review]
LGTM 🎉

File: pkg/rpcservice/middleware.go (hunk 1 above)
Comment (tomer-stripe) [reply pkg/rpcservice/middleware.go:49]: yep, good catch!

**Pull request merged**